### PR TITLE
feat(migrate,cli,sdk): add db drop command

### PR DESCRIPTION
**Pull request by Jolg42**
- drop
- Improved testing

### CODE REVIEW

File: src/packages/migrate/src/utils/ensureDatabaseExists.ts [hunk 1]
```diff
@@ -115,10 +115,36 @@ export async function askToCreateDb(
...
+      : credentials.type
+
+  const schemaWord = 'database'
+
+  return {
+    dbName,
+    dbType,
+    schemaWord,
+  }
+}
```
Comment (Jolg42) [inline src/packages/migrate/src/utils/ensureDatabaseExists.ts:150]: This part would need a MSSQL equivalent

File: src/packages/migrate/src/__tests__/DbPush.test.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
+process.env.GITHUB_ACTIONS = '1'
```
Comment (timsuchanek) [inline src/packages/migrate/src/__tests__/DbPush.test.ts:1]: A little comment why this is needed would be nice

File: src/packages/migrate/src/commands/DbDrop.ts [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+      if (json.message) {
+        throw Error(json.message)
+      }
+
+      throw Error(e)
+    }
+
+    if (
+      result?.exitCode === 0 &&
+      result.stderr.includes('The database was successfully dropped')
```
Comment (timsuchanek) [inline src/packages/migrate/src/commands/DbDrop.ts:151]: Just to double check, maybe you want to check for `result` to be there.
A simple `if(!result) { throw...}` can already be enough.

File: src/packages/migrate/src/utils/ensureDatabaseExists.ts (hunk 1 above)
Comment (timsuchanek) [reply src/packages/migrate/src/utils/ensureDatabaseExists.ts:150]: Yes please! We actually need that :) Can you add it?

File: src/packages/migrate/src/commands/DbDrop.ts (hunk 3 above)
Comment (Jolg42) [reply src/packages/migrate/src/commands/DbDrop.ts:151]: But result (execa response) is always there if engines returns something without error

File: src/packages/migrate/src/__tests__/DbPush.test.ts (hunk 2 above)
Comment (Jolg42) [reply src/packages/migrate/src/__tests__/DbPush.test.ts:1]: It's to get into the CI "mode" for snapshots, until we have some tests that can test both

File: src/packages/migrate/src/utils/ensureDatabaseExists.ts (hunk 1 above)
Comment (Jolg42) [reply src/packages/migrate/src/utils/ensureDatabaseExists.ts:150]: Done in https://github.com/prisma/prisma/commit/c0c70e96ae2ba8dfe669b4cf9ceefec5ac9937be

**Pull request merged**